library/LuaApi.cpp

```diff
@@ -1505,7 +1507,127 @@ static const LuaWrapper::FunctionReg dfhack_gui_module[] = {
     { NULL, NULL }
 };
 
+static int gui_autoDFAnnouncement(lua_State *state)
+{
+    int rv;
+    df::report_init *r = Lua::GetDFObject<df::report_init>(state, 1);
+
+    if (r)
+    {
+        std::string message = luaL_checkstring(state, 2);
+
+        if (lua_gettop(state) >= 3)
+            rv = Gui::autoDFAnnouncement(*r, message, lua_toboolean(state, 3));
+        else
+            rv = Gui::autoDFAnnouncement(*r, message);
+    }
+    else
+    {
+        df::coord pos;
+        int color;
+        bool bright, is_sparring, log_failures;
+        df::unit *unit1, *unit2;
+
```

Comment:
it looks like you need to initialize these to make the compiler happy. this is tripping `-Werror`.